src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc

```diff
@@ -0,0 +1,535 @@
+  }
+}
+
+void EventEngineDNSRequestWrapper::OnBalancerHostnamesResolved(
+    std::string authority,
+    absl::StatusOr<std::vector<EventEngine::ResolvedAddress>>
+        new_balancer_addresses) {
+  absl::optional<grpc_core::Resolver::Result> result;
+  {
+    MutexLock lock(&on_resolved_mu_);
+    if (orphaned_) return;
+    ++number_of_balancer_hostnames_resolved_;
+    if (new_balancer_addresses.ok()) {
+      if (!balancer_addresses_.ok()) {
+        GRPC_EVENT_ENGINE_DNS_TRACE(
+            "DNSResolver::%p previous balancer hostname lookup error, ignoring "
+            "%zu found hostnames.",
+            this, new_balancer_addresses->size());
+      }
+      balancer_addresses_->reserve(balancer_addresses_->size() +
+                                   new_balancer_addresses->size());
+      auto srv_channel_args =
+          resolver_->channel_args().Set(GRPC_ARG_DEFAULT_AUTHORITY, authority);
+      for (const auto& addr : *new_balancer_addresses) {
+        balancer_addresses_->emplace_back(CreateGRPCResolvedAddress(addr),
+                                          srv_channel_args);
+      }
+    } else {
+      auto tmp_status = balancer_addresses_.status();
+      grpc_core::StatusAddChild(&tmp_status, new_balancer_addresses.status());
+      balancer_addresses_ = tmp_status;
+    }
+    result = OnResolvedLocked();
+  }
+  if (result.has_value()) {
+    resolver_->OnRequestComplete(std::move(*result));
+  }
+}
+
+void EventEngineDNSRequestWrapper::OnTXTResolved(
+    absl::StatusOr<std::string> service_config) {
+  absl::optional<grpc_core::Resolver::Result> result;
+  {
+    MutexLock lock(&on_resolved_mu_);
+    if (orphaned_) return;
+    GPR_ASSERT(txt_handle_.has_value());
+    txt_handle_.reset();
+    service_config_json_ = service_config;
+    result = OnResolvedLocked();
+  }
+  if (result.has_value()) {
+    resolver_->OnRequestComplete(std::move(*result));
+  }
+}
+
+absl::Status
+EventEngineDNSRequestWrapper::GetResolutionFailureErrorMessageLocked() {
+  std::string error_msg =
+      absl::StrFormat("resolver:%p dns resolution failed.", this);
+  if (!addresses_.ok()) {
+    absl::StrAppend(&error_msg, " Hostname lookup error: ",
+                    grpc_core::StatusToString(addresses_.status()));
+  }
+  if (!balancer_addresses_.ok()) {
+    absl::StrAppend(&error_msg, " SRV lookup error: ",
+                    grpc_core::StatusToString(balancer_addresses_.status()));
+  }
+  if (!service_config_json_.ok()) {
+    absl::StrAppend(&error_msg, " TXT lookup error: ",
+                    grpc_core::StatusToString(service_config_json_.status()));
+  }
+  auto status = grpc_core::StatusCreate(
+      absl::StatusCode::kUnavailable, error_msg, DEBUG_LOCATION, /*children=*/
+      {addresses_.status(), balancer_addresses_.status(),
+       service_config_json_.status()});
+  return status;
+}
+
+void EventEngineDNSRequestWrapper::MaybePopulateAddressesLocked(
+    grpc_core::Resolver::Result& result) {
```

Comment:
Please pass in mutable parameters as pointers instead of references.

Same thing throughout.

Reply:
I went with the mutable reference choice based on totw/178.

Reply:
That's unfortunate.  I guess at some point we should consider updating to that style, no matter how sad it makes me. :(